datafusion/expr/src/window_function.rs

```diff
@@ -32,7 +32,7 @@ use strum_macros::EnumIter;
 
 /// WindowFunction
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum WindowFunction {
+pub enum WindowFunctionDefinition {
```

Comment:
can we move `WindowFunctionDefinition` into `datafusion/expr/src/expr.rs`, make this consist of `ScalarFunctionDefinition` and `AggregateFunctionDefinition` .

Reply:
If I understood correctly, this [2f4ce8368ae5003648919bb1201f08a2dbe23d5f](https://github.com/apache/arrow-datafusion/pull/8382/commits/2f4ce8368ae5003648919bb1201f08a2dbe23d5f) should be it. There are two commits after this, one to merge a change in expr.rs in the main branch and the following one to reformat

Reply:
@edmondop sorry for make you misunderstood, I mean we only move the code related to `WindowFunctionDefinition`
https://github.com/apache/arrow-datafusion/blob/e19c669855baa8b78ff86755803944d2ddf65536/datafusion/expr/src/window_function.rs#L34-L44
https://github.com/apache/arrow-datafusion/blob/e19c669855baa8b78ff86755803944d2ddf65536/datafusion/expr/src/window_function.rs#L69-L78
https://github.com/apache/arrow-datafusion/blob/e19c669855baa8b78ff86755803944d2ddf65536/datafusion/expr/src/window_function.rs#L166-L178
https://github.com/apache/arrow-datafusion/blob/e19c669855baa8b78ff86755803944d2ddf65536/datafusion/expr/src/window_function.rs#L227-L237
into `datafusion/expr/src/expr.rs`.